consensus/src/epoch_manager.rs

```diff
@@ -898,7 +898,7 @@ impl<P: OnChainConfigProvider> EpochManager<P> {
             pipeline_backpressure_config,
             chain_health_backoff_config,
             self.quorum_store_enabled,
-            onchain_consensus_config.validator_txn_enabled(),
+            onchain_consensus_config.effective_validator_txn_config(),
```

Comment:
what does effective mean here?

Reply:
It means sometimes `onchain_consensus_config` doesn't hold such object directly. But it is its job to return one, and here it's coming up with one that matches itself